internal/clients/telegram/state/daily_status.go

```diff
@@ -0,0 +1,235 @@
+	"github.com/m-kuzmin/daily-reporter/internal/clients/telegram/update"
+	"github.com/m-kuzmin/daily-reporter/internal/template"
+	"github.com/m-kuzmin/daily-reporter/internal/util/option"
+	"github.com/m-kuzmin/daily-reporter/internal/util/slashcmd"
+	"github.com/pkg/errors"
+)
+
+const dailyStatusItemLimit = 100
+
+type DailyStatusHandler struct {
+	responses *DailyStatusResponses
+	userData  UserSharedData
+	DailyStatusState
+}
+
+func (s *DailyStatusHandler) GroupTextMessage(message update.GroupTextMessage) Transition {
+	return s.handleDailyStatus(message.Chat.ID, message.Text)
+}
+
+func (s *DailyStatusHandler) PrivateTextMessage(message update.PrivateTextMessage) Transition {
+	return s.handleDailyStatus(message.Chat.ID, message.Text)
+}
+
+func (s *DailyStatusHandler) CallbackQuery(callback update.CallbackQuery) Transition {
+	ignoreWithAlert := NewTransition(s.RootState, s.userData, []response.BotAction{
+		response.CallbackQueryAnswerNotification(callback, "This button doesnt work."),
+	})
+
+	data, isSome := callback.Data.Unwrap()
+	if !isSome {
+		return ignoreWithAlert
+	}
+
+	message, isSome := callback.Message.Unwrap()
+	if !isSome {
+		return ignoreWithAlert
+	}
+
+	if transition, handled := s.handleCQSaveDefaultProject(data, message); handled {
+		return transition
+	}
+
+	return ignoreWithAlert
+}
+
+func (s *DailyStatusHandler) Ignore() Transition {
+	return NewTransition(s.DailyStatusState, s.userData, response.Nothing())
+}
+
+//nolint:cyclop
+func (s *DailyStatusHandler) handleDailyStatus(chatID update.ChatID, text string) Transition {
+	cmd, isCmd := slashcmd.Parse(text)
+
+	if isCmd && strings.ToLower(cmd.Method) == cancelCommand {
+		return NewTransition(s.RootState, s.userData, []response.BotAction{
+			response.NewSendMessage(response.ChatID(fmt.Sprint(chatID)), "Canceled."),
+		})
+	}
+
+	switch s.Stage {
+	case ignoreMessagesDailyStatusStage:
+		break
+
+	case discoveryOfTheDayDailyStatusStage:
+		s.DailyStatusState.Stage = questionsAndBlockersDailyStatusStage
+
+		if isCmd && strings.ToLower(cmd.Method) == noneCommand {
+			s.DiscoveryOfTheDay = option.None[string]()
+		} else {
+			s.DiscoveryOfTheDay = option.Some(text)
+		}
+
+		return NewTransition(s.DailyStatusState, s.userData, []response.BotAction{
+			response.NewSendMessage(response.ChatID(fmt.Sprint(chatID)),
+				s.responses.QuestionsAndBlockers,
+			),
+		})
+
+	case questionsAndBlockersDailyStatusStage:
+		if isCmd && strings.ToLower(cmd.Method) == noneCommand {
+			s.QuestionsAndBlockers = option.None[string]()
+		} else {
+			s.QuestionsAndBlockers = option.Some(text)
+		}
+
+		report, err := s.generateReport()
+		if err != nil {
+			report = github.GqlErrorStringOr("Github API error: %s", err, "Something went wrong while contacting GitHub.")
+		}
+
+		return NewTransition(s.RootState, s.userData, []response.BotAction{
+			response.NewSendMessage(response.ChatID(fmt.Sprint(chatID)), report),
+		})
+	}
+
+	return s.Ignore()
```

Comment:
It seems the break can be replaced wit this return

https://github.com/m-kuzmin/daily-reporter/pull/7/files#diff-2f46d1c13500daa8966c5f71dc49b78bf584ada8c961fa953941e09b3fed51f5R72